src/main/java/org/opensearch/agent/tools/RAGTool.java

```diff
@@ -52,6 +52,7 @@ public class RAGTool extends AbstractRetrieverTool {
     public static final String EMBEDDING_MODEL_ID_FIELD = "embedding_model_id";
     public static final String EMBEDDING_FIELD = "embedding_field";
     public static final String OUTPUT_FIELD = "output_field";
+    public static final String QUERY_TYPE = "query_type";
```

Comment:
Since we only use RAGTool in Olly, we may need a new parameter to control whether we directly return search results or call LLM to do summarize.

Reply:
I mean we need a parameter to control whether to invoke the MLPredictionTaskAction, not how to interact with another model. Maybe users don't need further post-process for search results, but in current code we always do execute MLPredictionTaskAction after search. And we need a parameter decide whether RAGTool need to do this.

Reply:
added a parameter [enableContentGeneration to RAGTool](https://github.com/opensearch-project/skills/pull/140/commits/b12fd8ad2c9196f89d1978df7ba05dfdbf05b491)